src/internal/RSocketConnectionManager.cpp

```diff
@@ -87,8 +88,20 @@ void RSocketConnectionManager::manageConnection(
       *this, socket, eventBase);
   newNetworkStats->inner = std::move(socket->networkStats());
   socket->networkStats() = std::move(newNetworkStats);
+  sockets_.lock()->insert({std::move(socket), {eventBase, resumeToken}});
```

Comment:
Right. Since most applications are going to repeat the same logic, I felt it is better to embed this in the protocol implementation.  The application wouldn't have to know about StateMachines and ResumeTokens.  May be we could keep this logic in the protocol implementation, but the application can override it if needed ? Anyway, eagerly looking forward to see what you have in your mind.